core/templates/pages/topic-viewer-page/topic-viewer-page.controller.ts

```diff
@@ -54,28 +55,40 @@ angular.module('oppia').directive('topicViewerPage', [
           ctrl.setActiveTab = function(newActiveTabName) {
             ctrl.activeTab = newActiveTabName;
           };
+          ctrl.getStaticImageUrl = function(imagePath) {
+            return UrlInterpolationService.getStaticImageUrl(imagePath);
+          };
           ctrl.checkMobileView = function() {
             return (WindowDimensionsService.getWidth() < 500);
           };
           ctrl.$onInit = function() {
-            ctrl.canonicalStoriesList = [];
-            ctrl.setActiveTab('story');
+            ctrl.canonicalStorySummaries = [];
+            ctrl.setActiveTab('info');
             ctrl.topicName = UrlService.getTopicNameFromLearnerUrl();
 
             PageTitleService.setPageTitle(ctrl.topicName + ' - Oppia');
 
             LoaderService.showLoadingScreen('Loading');
+            ctrl.topicIsLoading = true;
             TopicViewerBackendApiService.fetchTopicData(ctrl.topicName).then(
               function(readOnlyTopic) {
                 ctrl.topicId = readOnlyTopic.getTopicId();
-                ctrl.canonicalStoriesList = (
+                ctrl.topicName = readOnlyTopic.getTopicName();
+                ctrl.topicDescription = readOnlyTopic.getTopicDescription();
+                ctrl.canonicalStorySummaries = (
                   readOnlyTopic.getCanonicalStorySummaries());
+                ctrl.chapterCount = 0;
+                for (var idx in ctrl.canonicalStorySummaries) {
+                  ctrl.chapterCount += (
+                    ctrl.canonicalStorySummaries[idx].getNodeTitles().length);
+                }
                 ctrl.degreesOfMastery = readOnlyTopic.getDegreesOfMastery();
                 ctrl.subtopics = readOnlyTopic.getSubtopics();
                 ctrl.skillDescriptions = readOnlyTopic.getSkillDescriptions();
                 LoaderService.hideLoadingScreen();
                 ctrl.trainTabShouldBeDisplayed = (
                   readOnlyTopic.getTrainTabShouldBeDisplayed());
+                ctrl.topicIsLoading = false;
```

Comment:
considering moving this above LoaderService.hideLoadingScreen(); on line 88 if it effects layout

Reply:
Done